kdc/krb5tgs.c

```diff
@@ -814,12 +835,17 @@ tgs_make_reply(krb5_context context,
     ret = copy_Realm(&tgt_name->realm, &rep.crealm);
     if (ret)
 	goto out;
-/*
-    if (f.request_anonymous)
-	_kdc_make_anonymous_principalname (&rep.cname);
-    else */
 
-    ret = copy_PrincipalName(&tgt_name->name, &rep.cname);
+    /*
+     * RFC 8062 states "if the ticket in the TGS request is an anonymous
+     * one, the client and client realm are copied from that ticket". So
+     * whilst the TGT flag check below is superfluous, it is included in
+     * order to follow the specification to its letter.
+     */
+    if (et.flags.anonymous && !tgt->flags.anonymous)
+	_kdc_make_anonymous_principalname(&rep.cname);
```

Comment:
The comment should instead explain the replacement of the client name with the anonymous name when an authenticated client wants an anonymous ticket.  The normal case barely needs commenting. (Not a release blocker).